### DISCUSSION TITLE
wiki.service not working (?) - debian

### DISCUSSION THREAD

**Discussion by ArnaudTe** [requarks/wiki#7256] [Help / Questions] [answered]
Hi,

For some reason i have to manually do node server to be able to access the wiki.
my wiki.service is the same as in https://docs.requarks.io/install/linux and my node version is 16.2. 

I don't know what i should look for ? wiki status shows up as working with no error and everything works fine after i manually do node server.

Thanks ! 👍

**Comment by fnafox** [comment]
check file permissions

**Reply by ArnaudTe** [reply]
i've checked file permission on the /var/wiki and they are the same as before ?
is there an other place i need to check ?

**Reply by fnafox** [reply]
my best guess is that some permissions or environment is broken if you can run it manually but the service doesn't work. there is nothing fundamentally different happening when the service starts the application compared to you doing it manually.

assuming your service is called wiki, what does `journalctl -xeu wiki` say?

**Reply by ArnaudTe** [reply]
` L'unité (unit) wiki.service a terminé son démarrage, avec le résultat done.
juil. 17 11:55:18 SRV-WIKI node[514]: Loading configuration from /var/wiki/config.yml... OK
juil. 17 11:55:18 SRV-WIKI node[514]: 2024-07-17T09:55:18.863Z [MASTER] info: =======================================
juil. 17 11:55:18 SRV-WIKI node[514]: 2024-07-17T09:55:18.864Z [MASTER] info: = Wiki.js 2.5.303 =====================
juil. 17 11:55:18 SRV-WIKI node[514]: 2024-07-17T09:55:18.864Z [MASTER] info: =======================================
juil. 17 11:55:18 SRV-WIKI node[514]: 2024-07-17T09:55:18.865Z [MASTER] info: Initializing...
juil. 17 11:55:19 SRV-WIKI node[514]: 2024-07-17T09:55:19.428Z [MASTER] info: Using database driver pg for postgres [ OK ]
juil. 17 11:55:19 SRV-WIKI node[514]: 2024-07-17T09:55:19.444Z [MASTER] info: Connecting to database...
juil. 17 11:55:19 SRV-WIKI node[514]: 2024-07-17T09:55:19.479Z [MASTER] info: Database Connection Successful [ OK ]
juil. 17 11:55:20 SRV-WIKI node[514]: 2024-07-17T09:55:20.057Z [MASTER] warn: Mail is not setup! Please set the configuration in the administra>
juil. 17 11:55:20 SRV-WIKI node[514]: 2024-07-17T09:55:20.151Z [MASTER] info: Loading GraphQL Schema...
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.175Z [MASTER] info: GraphQL Schema: [ OK ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.346Z [MASTER] info: HTTP Server on port: [ 3000 ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.367Z [MASTER] info: HTTPS Server on port: [ 443 ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.378Z [MASTER] info: HTTP Server: [ RUNNING ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.379Z [MASTER] warn: listen EACCES: permission denied 0.0.0.0:443
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.480Z [MASTER] info: No new analytics providers found: [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.522Z [MASTER] info: Loaded 21 authentication strategies: [ OK ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.530Z [MASTER] info: No new comment providers found: [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.546Z [MASTER] info: No new editors found: [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.568Z [MASTER] info: No new loggers found: [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.608Z [MASTER] info: No new renderers found: [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.630Z [MASTER] info: No new search engines found: [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.661Z [MASTER] info: No new storage targets found: [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.662Z [MASTER] info: Checking for installed optional extensions...
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.672Z [MASTER] info: Optional extension git was not found on this system. [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.677Z [MASTER] info: Optional extension pandoc was not found on this system. [ SKIPPED>
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.683Z [MASTER] info: Optional extension puppeteer was not found on this system. [ SKIP>
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.684Z [MASTER] info: Optional extension sharp was not found on this system. [ SKIPPED ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.689Z [MASTER] info: Authentication Strategy Local: [ OK ]
juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.770Z [MASTER] info: Authentication Strategy LDAP / Active Directory: [ OK ]
juil. 17 11:55:22 SRV-WIKI node[514]: 2024-07-17T09:55:22.532Z [MASTER] info: (COMMENTS/DEFAULT) Initializing...
juil. 17 11:55:22 SRV-WIKI node[514]: 2024-07-17T09:55:22.533Z [MASTER] info: (COMMENTS/DEFAULT) Initialization completed.
juil. 17 11:55:22 SRV-WIKI node[514]: 2024-07-17T09:55:22.547Z [MASTER] info: Purging orphaned upload files...
juil. 17 11:55:22 SRV-WIKI node[514]: 2024-07-17T09:55:22.556Z [MASTER] info: Syncing locales with Graph endpoint...
juil. 17 11:55:22 SRV-WIKI node[514]: 2024-07-17T09:55:22.560Z [MASTER] info: Fetching latest updates from Graph endpoint...
juil. 17 11:55:22 SRV-WIKI node[514]: 2024-07-17T09:55:22.588Z [MASTER] info: Telemetry is active: [ OK ]
juil. 17 11:55:22 SRV-WIKI node[514]: 2024-07-17T09:55:22.592Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 11:55:22 SRV-WIKI node[663]: Loading configuration from /var/wiki/config.yml... OK
juil. 17 11:55:22 SRV-WIKI node[663]: 2024-07-17T09:55:22.770Z [JOB] info: Rebuilding page tree...
juil. 17 11:55:23 SRV-WIKI node[514]: 2024-07-17T09:55:23.002Z [MASTER] info: Fetching latest updates from Graph endpoint: [ COMPLETED ]
juil. 17 11:55:23 SRV-WIKI node[663]: 2024-07-17T09:55:23.133Z [JOB] info: Using database driver pg for postgres [ OK ]
juil. 17 11:55:23 SRV-WIKI node[514]: 2024-07-17T09:55:23.189Z [MASTER] info: Pulled latest locale updates for French from Graph endpoint: [ CO>
juil. 17 11:55:23 SRV-WIKI node[663]: 2024-07-17T09:55:23.213Z [JOB] info: Rebuilding page tree: [ COMPLETED ]
juil. 17 11:55:23 SRV-WIKI node[514]: 2024-07-17T09:55:23.346Z [MASTER] info: Pulled latest locale updates for English from Graph endpoint: [ C>
juil. 17 11:55:23 SRV-WIKI node[514]: 2024-07-17T09:55:23.351Z [MASTER] info: Syncing locales with Graph endpoint: [ COMPLETED ]
juil. 17 12:10:22 SRV-WIKI node[514]: 2024-07-17T10:10:22.596Z [MASTER] info: Purging orphaned upload files...
juil. 17 12:10:22 SRV-WIKI node[514]: 2024-07-17T10:10:22.598Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 12:25:22 SRV-WIKI node[514]: 2024-07-17T10:25:22.599Z [MASTER] info: Purging orphaned upload files...
juil. 17 12:25:22 SRV-WIKI node[514]: 2024-07-17T10:25:22.600Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 12:40:22 SRV-WIKI node[514]: 2024-07-17T10:40:22.600Z [MASTER] info: Purging orphaned upload files...
juil. 17 12:40:22 SRV-WIKI node[514]: 2024-07-17T10:40:22.601Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 12:55:22 SRV-WIKI node[514]: 2024-07-17T10:55:22.602Z [MASTER] info: Purging orphaned upload files...
juil. 17 12:55:22 SRV-WIKI node[514]: 2024-07-17T10:55:22.603Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 13:10:22 SRV-WIKI node[514]: 2024-07-17T11:10:22.605Z [MASTER] info: Purging orphaned upload files...
juil. 17 13:10:22 SRV-WIKI node[514]: 2024-07-17T11:10:22.606Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 13:25:22 SRV-WIKI node[514]: 2024-07-17T11:25:22.606Z [MASTER] info: Purging orphaned upload files...
juil. 17 13:25:22 SRV-WIKI node[514]: 2024-07-17T11:25:22.608Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 13:40:22 SRV-WIKI node[514]: 2024-07-17T11:40:22.609Z [MASTER] info: Purging orphaned upload files...
juil. 17 13:40:22 SRV-WIKI node[514]: 2024-07-17T11:40:22.610Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 13:55:22 SRV-WIKI node[514]: 2024-07-17T11:55:22.610Z [MASTER] info: Purging orphaned upload files...
juil. 17 13:55:22 SRV-WIKI node[514]: 2024-07-17T11:55:22.611Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 14:10:22 SRV-WIKI node[514]: 2024-07-17T12:10:22.613Z [MASTER] info: Purging orphaned upload files...
juil. 17 14:10:22 SRV-WIKI node[514]: 2024-07-17T12:10:22.614Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 14:25:22 SRV-WIKI node[514]: 2024-07-17T12:25:22.614Z [MASTER] info: Purging orphaned upload files...
juil. 17 14:25:22 SRV-WIKI node[514]: 2024-07-17T12:25:22.616Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 14:40:22 SRV-WIKI node[514]: 2024-07-17T12:40:22.617Z [MASTER] info: Purging orphaned upload files...
juil. 17 14:40:22 SRV-WIKI node[514]: 2024-07-17T12:40:22.618Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 14:55:22 SRV-WIKI node[514]: 2024-07-17T12:55:22.618Z [MASTER] info: Purging orphaned upload files...
juil. 17 14:55:22 SRV-WIKI node[514]: 2024-07-17T12:55:22.619Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]
juil. 17 15:10:22 SRV-WIKI node[514]: 2024-07-17T13:10:22.619Z [MASTER] info: Purging orphaned upload files...
juil. 17 15:10:22 SRV-WIKI node[514]: 2024-07-17T13:10:22.620Z [MASTER] info: Purging orphaned upload files: [ COMPLETED ]

root@SRV-WIKI:~#`

It doesn't say much to me but may be u can understand something ?

**Reply by fnafox** [reply, marked as answer]
` juil. 17 11:55:21 SRV-WIKI node[514]: 2024-07-17T09:55:21.379Z [MASTER] warn: listen EACCES: permission denied 0.0.0.0:443`
Wiki is running correctly, but you're running your server on port 443 which the user doesn't have permissions to bind to. Ports under 1000 are protected.

Add the following line into your wiki.service file under the [Service] tag:
```
[Service]
AmbientCapabilities=CAP_NET_BIND_SERVICE
```

Then run
```
systemctl daemon-reload
systemctl restart wiki
```

**Reply by ArnaudTe** [reply]
Okay i though the warning was not important, it was indeed. 
Does that mean i have the rights to this when i launch it manually ?
Thank you !

**Reply by fnafox** [reply]
That depends how you were launching it. If you were doing it as root user (e.g. if using sudo) then you had permissions.
